### PR TITLE
[11.0][ADD] Module which add a redirection button to the project from the SO

### PR DESCRIPTION
**Pull request by benwillig**
Simply adds a button on the sale order to be redirected on the linked projects.
Linked projects are:

- Projects linked to order's analytic account
- Projects linked to order line's tasks
- Projects linked to the order line's product

### CODE REVIEW

File: sale_timesheet_project_link/models/sale_order.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+    @api.multi
+    @api.depends(
+        'order_line.task_id.project_id',
+        'project_ids',
+    )
+    def _compute_all_projects(self):
+        for rec in self:
+            rec.all_project_ids = (
+                rec.order_line.mapped('task_id.project_id') | rec.project_ids)
```
Comment (sbidoul) [inline sale_timesheet_project_link/models/sale_order.py]: Is a computed field the best way to do this? A simple method would be less stressful for the orm cache.

File: sale_timesheet_project_link/README.rst [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+===========================
+Sale Timesheet Project Link
+===========================
+
+Simply adds a button on the sale order to be redirected on the linked projects.
+Linked projects are:
+
+#. Projects linked to order's analytic account
+#. Projects linked to order line's tasks
+#. Projects linked to the order line's product
```
Comment (sbidoul) [inline sale_timesheet_project_link/README.rst:14]: I don't see this last item in _compute_all_projects, is it obtained indirectly?

Comment (benwillig) [reply sale_timesheet_project_link/README.rst:14]: The last item is given by the ```project_id``` field available in the ```sale_timesheet``` module.

File: sale_timesheet_project_link/models/sale_order.py (hunk 1 above)
Comment (benwillig) [reply sale_timesheet_project_link/models/sale_order.py]: Yes I don't really need this field, it is only used when counting the number of projects to show the button in the form view. I will change it for a simple method.

**Comment by rafaelbn** [comment]
Thanks @benwillig ! Please could you review travis?

**Comment by benwillig** [comment]
The travis error is due to the ```sale_automatic_workflow``` module and this is not caused by this PR. Unless I missed something.

**Comment by pedrobaeza** [comment]
Main branch is green. Please rebase to see if it helps.

**Comment by benwillig** [comment]
Indeed, everything is green now.

**Review by rousseldenis — APPROVED** [review]
Code review

**Pull request merged**